lib/bootloader.js

```diff
@@ -45,7 +45,12 @@
   // version without applicationCache intercepting us.
   function getShadowIndex(success, failure) {
     var req = new XMLHttpRequest();
-    req.open('GET', 'APPSHELL_PATHappshell.html', true);
+    var path = 'APPSHELL_PATHappshell';
+    var query = window.location.search.match(/revision=([^&#]+)/);
```

Comment:
Read the PR description.

Case:
1. Load app with current revision, appcache caches
2. Load app with revision param
   - appcache is USED to boot app (current revision)
   - bootloader finds no manifest, says OBSOLETE, not any of the above
   - DOM swap happens